### PR TITLE
Windows/Vulkan: implement screenshots. 

### LINKED ISSUE #10134: Screenshot feature broken in Vulkan
It crashes in VKContext::CopyFramebufferToMemorySync with a null source buffer.

### PR DESCRIPTION
**Pull request by hrydgard**
Something is iffy with the red/blue swap though...

Fixes #10134 .

### CODE REVIEW

**Comment by LunaMoo** [comment]
Works nicely, thanks.:)

File: ext/native/thin3d/thin3d.cpp [hunk 1]
```diff
@@ -381,6 +381,31 @@ void ConvertFromRGBA8888(uint8_t *dst, const uint8_t *src, uint32_t dstStride, u
...
 
+// TODO: SSE/NEON
+// Could also make C fake-simd for 64-bit, two 8888 pixels fit in a register :)
+void ConvertFromBGRA8888(uint8_t *dst, const uint8_t *src, uint32_t dstStride, uint32_t srcStride, uint32_t width, uint32_t height, DataFormat format) {
+	// Must skip stride in the cases below.  Some games pack data into the cracks, like MotoGP.
+	const uint32_t *src32 = (const uint32_t *)src;
+
+	if (format == Draw::DataFormat::R8G8B8A8_UNORM) {
+		uint32_t *dst32 = (uint32_t *)dst;
+		if (src == dst) {
```
Comment (unknownbrackets) [inline ext/native/thin3d/thin3d.cpp]: Well, presumably we can't skip in this case, right?  I guess it won't happen, though.

-[Unknown]

File: ext/native/thin3d/thin3d.cpp [hunk 2]
```diff
@@ -381,6 +381,31 @@ void ConvertFromRGBA8888(uint8_t *dst, const uint8_t *src, uint32_t dstStride, u
...
+			return;
+		}
+		else {
+			for (uint32_t y = 0; y < height; ++y) {
+				ConvertBGRA8888ToRGBA8888(dst32, src32, width);
+				memcpy(dst32, src32, width * 4);
```
Comment (unknownbrackets) [inline ext/native/thin3d/thin3d.cpp]: Hmm, memcpy after the `ConvertBGRA8888ToRGBA8888` can't be right - this is probably why the BGRA hack?

-[Unknown]

**Review by unknownbrackets — COMMENTED** [review]
Ah, I tried something similar but didn't think about adding TRANSFER_SRC, I see.  Mistakenly assumed that wasn't an option...

-[Unknown]

File: ext/native/thin3d/thin3d.cpp (hunk 2 above)
Comment (hrydgard) [reply ext/native/thin3d/thin3d.cpp]: Oh, haha. Thanks for spotting!

**Pull request merged**